### PR TITLE
#27 #28 | API Route for POST & GET user detail

### PR DESCRIPTION
**Pull request by Khukhuna**
Part of GCI 18 :tada: :heart: 

[Task](https://codein.withgoogle.com/dashboard/task-instances/5725239146184704/): Create API Route for POST user details-Lead Management Server
[Task](https://codein.withgoogle.com/dashboard/task-instances/6319837304324096/): Setup and Fix a issue during setup [lead management server]

___
- **URL**:
  - **POST** domain.org/api/user
  - **GET** domain.org/api/user/:number

___

- **Body parameters for POST request**:
   - username
  - phone number
  - occupation
  - is current customer or not
- **Parameters for GET request**:
  - phone number
___

If this pull request will be merged, It's necessary to delete database connection in index.js  (from line 13 to line 20), because setting up database connection is another task.

_**Example of sending POST request to API:**_

### CODE REVIEW

**Comment by Priya997** [comment]
Good Work :D

File: index.js [hunk 1]
```diff
@@ -4,12 +4,44 @@ const bodyParser = require("body-parser");
...
 const http = require('http').Server(server);
 const io = require('socket.io')(http);
+const mongoose = require('mongoose');
+
+const userController = require('./controllers/user.controller.js');
+
+
+// THIS SHOULD BE DELETED
+mongoose.Promise = global.Promise;
+mongoose.connect("mongodb://test:test123@ds239873.mlab.com:39873/lead-management-server", {
```
Comment (nikitadhiman) [inline index.js]: Good attempt but don't push url directly

**Comment by Khukhuna** [comment]
You are right :smiley:
I'll make that change.

**Comment by Khukhuna** [comment]
@jatinkatyal13 Should we respond with this body?
```json
{
    "is_customer": true,
    "_id": "5bd34bfc9d41461c5a4074bd",
    "username": "Saba Khukhunashvili",
    "phone_number": "555123123",
    "occupation": "Georgian",
    "__v": 0
}
```
it will be easier to get the data on Android side (Retrofit will not need another pojo).

**Comment by jatinkatyal13** [comment]
Amazing 🚀 👍

**Comment by divyanshkul** [comment]
Reopening this for now, as the student merged two tasks into one.
@Khukhuna  please do different tasks in different branches and make separate PRs.
Please make changes as discussed in the gitter chat.

**Comment by divyanshkul** [comment]
Closing this
Good work
**Mentor note:** this one is good to go for the merge.

**Pull request closed without merging**